### PR TITLE
Fix #238 branch for region/voice server changes

### LINKED ISSUE #238: Voice: v2: ReadPacket() never finishes when region changes
When the voice server's region changes, `ReadPacket()` will never finish as it (presumably) waits indefinitely to read a packet from the old UDP voice server.

I believe the solution here is to `SetReadDeadline` to something like 1-2 seconds on the underlying connection, read, check if the gateway has changed, otherwise re-attempt the read and loop.

### PR DESCRIPTION
**Pull request by 1lann**
I tested that this works, region changes are handled gracefully with packets from `ReadPacket` resuming like nothing happened.

### CODE REVIEW

File: voice/udp/manager.go [hunk 1]
```diff
@@ -160,7 +165,13 @@ func (m *Manager) ReadPacket() (p *Packet, err error) {
 		return nil, ErrManagerClosed
 	}
 
-	return conn.ReadPacket()
+	p, err = conn.ReadPacket()
+	if err != nil && conn.IsClosed() {
+		wsutil.WSDebug("UDP connection was closed, re-attempting read...")
+		return m.ReadPacket()
```
Comment (diamondburned) [inline voice/udp/manager.go]: I think it's a better idea to make this a loop. Using recursion for this seems overkill.

File: voice/udp/manager.go [hunk 2]
```diff
@@ -177,14 +188,9 @@ func (m *Manager) Write(b []byte) (n int, err error) {
 // acquireConn acquires the current connection and releases the lock, returning
 // the connection at that point in time. Nil is returned if Manager is closed.
 func (m *Manager) acquireConn() *Connection {
-	m.stopMu.Lock()
-	defer m.stopMu.Unlock()
-
 	select {
 	case m.connLock <- struct{}{}:
```
Comment (diamondburned) [inline voice/udp/manager.go]: This should probably be changed to just

```go
m.connLock <- struct{}{}
defer func() { <-m.connLock }()
```

File: voice/udp/manager.go [hunk 3]
```diff
@@ -127,6 +128,10 @@ func (m *Manager) PauseAndDial(ctx context.Context, addr string, ssrc uint32) (*
 		conn.ResetFrequency(m.frequency, m.timeIncr)
 	}
 
+	if m.conn != nil {
```
Comment (diamondburned) [inline voice/udp/manager.go]: Is there a reason why this is done after connecting and not before? It seems to me that `connLock` is acquired anyway, so there's no reason to not close it before.

**Review by diamondburned — CHANGES_REQUESTED** [review]

File: voice/udp/manager.go (hunk 1 above)
Comment (1lann) [reply voice/udp/manager.go]: tfw no tail call optimizatoin, sure.

File: voice/udp/manager.go (hunk 3 above)
Comment (1lann) [reply voice/udp/manager.go]: ¯\\\_(ツ)\_/¯ sounds good to me

**Review by diamondburned — APPROVED** [review]

**Pull request merged**